internals-js/src/coreSpec.ts

```diff
@@ -144,33 +149,129 @@ export type CoreImport = {
   as?: string,
 };
 
-export function extractCoreFeatureImports(directive: Directive<SchemaDefinition, CoreOrLinkDirectiveArgs>): CoreImport[] {
+export function extractCoreFeatureImports(url: FeatureUrl, directive: Directive<SchemaDefinition, CoreOrLinkDirectiveArgs>): CoreImport[] {
   const args = directive.arguments();
   if (!('import' in args)) {
     return [];
   }
-  const importArg = args.import;
-  const imports: CoreImport[] = importArg ? importArg.map((a) => typeof a === 'string' ? { name: a } : a) : [];
+  // Note: up to this point, we've kind of cheated with typing and force-casted `importArg` to a list of (string | CoreImport) (for convenience)
+  // but because `link__Import` is a scalar, there is no graphQL-side validation of the values and we need to manually check that is is indeed
+  // either a string or a valid `CoreImport` value.
+  const { imports, errors } = validateCoreFeatureImports(url, directive, args.import);
   for (const i of imports) {
     if (!i.as) {
       continue;
     }
     if (i.name.charAt(0) === '@' && i.as.charAt(0) !== '@') {
-      throw ERRORS.INVALID_LINK_DIRECTIVE_USAGE.err({
-        message: `Invalid @link import renaming: directive ${i.name} imported name should starts with a '@' character, but got "${i.as}"`,
+      errors.push(ERRORS.INVALID_LINK_DIRECTIVE_USAGE.err({
```

Comment:
**question** Is there a reason these checks are in this method vs `validateCoreFeatureImports`?

Reply:
Yeah, not really. I initially split `validateCoreFeatureImport` to only do the "syntax" validation, but it ended up doing more than that and so I agree having those validation at a different place is now just weird. I've merge those 2 methods together (once you move those checks in `validateCoreFeatureImports`, not much logic remains so ...).